File: funNLearn/src/main/java/dsAlgo/sort/BubbleSort.py
"""
    Tag: sort
"""
from typing import List


class Solution:
    def bubble_sort(self, A: List[int]) -> List[int]:
        for i in range(len(A) - 1):
            for j in range(i, len(A)):
                if A[i] > A[j]:
                    A[i], A[j] = A[j], A[i]
        return A


assert Solution().bubble_sort([3, 2, 9, 3, 5, 7, 1, 8]) == [1, 2, 3, 3, 5, 7, 8, 9]
print("Tests Passed!")

File: funNLearn/src/main/java/dsAlgo/sort/BubbleSort.java
/**
 * Tag: sort
 * 
 */

package dsAlgo.sort;

public class BubbleSort {
   int arr[] = { 2, 7, 6, 4, 1, 8, 5, 3 };

   public void bSort() {
      /* from zeroth element to n-1th element */
      for (int i = 0; i < arr.length - 2; i++) {
         // from 1st to arr.length-1th element compare
         // every adjacent pair and swap if left > right
         for (int j = 1; j < arr.length - i; j++) {
            if (arr[j - 1] > arr[j]) {
               int temp = arr[j - 1];
               arr[j - 1] = arr[j];
               arr[j] = temp;
            }
         }
      }
   }

   public static void main(String args[]) {
      BubbleSort ss = new BubbleSort();
      ss.bSort();
      for (int i = 0; i < ss.arr.length; i++)
         System.out.print(ss.arr[i] + ", ");
   }
}